site/content/blog/tilt-carvel-local-workflow.md

```diff
@@ -0,0 +1,202 @@
+Tilt is configured using the `Tiltfile` in the root of the project. Visual Studio code has a [plugin](https://marketplace.visualstudio.com/items?itemName=tilt-dev.Tiltfile)
+for editing Tiltfiles and is highly recommended.
+
+### Build stage
+First, a `local_resource` is defined to compile the Go application. A `local_resource` describes a task that runs on the local machine
+
+```python
+compile_cmd = """
+    rm build/carvel-tilt || true &&
+	go build \
+        -o build/carvel-tilt \
+        cmd/carvel-tilt/main.go &&
+        echo "Go build finished\n"
+"""
+local_resource(
+    'go-compile',
+    compile_cmd,
+    deps=['pkg', 'cmd', 'deployments'],
+    env={'GOOS': 'linux', 'GOARCH': 'amd64'}
+)
+
+```
+`compile_cmd` is a variable containing the go build command to run
+
+`local_resouce` describes how to run the command. The `deps` argument references the files or folders that will trigger
+this command to run.
+
+### Deploy stage
+To deploy the app to Kubernetes, `k8s_custom_deploy` is defined. This function runs a command that applies Kubernetes resources
+and then returns the created resources as YAML. `kapp inspect` is used to return the raw objects for Tilt.
+The `deps` argument is set to `build/carvel-tilt` as this forces Tilt to always run the build stage before the 
+deploy stage.
+
+```python
+port = 8084
+registry = os.getenv('REGISTRY','docker.io/ojhughes')
+
+kapp_apply_cmd = """
+    ytt --file deployments --data-value-yaml port=%d -v registry=%s| 
+    kbld -f - | 
+    kapp deploy -n default -a carvel-tilt-demo -y -f - > /dev/null &&
+    kapp inspect -n default -a carvel-tilt-demo --raw --tty=false
+""" % (port, registry)
+
+kapp_delete_cmd = "kapp delete -n default -a carvel-tilt-demo -y"
+
+k8s_custom_deploy(
+    name='carvel-tilt-demo',
+    deps=['build/carvel-tilt'],
+    apply_cmd=kapp_apply_cmd,
+    delete_cmd=kapp_delete_cmd
+)
+k8s_resource('carvel-tilt-demo', port_forwards=port, auto_init=False )
+```
+### Port forwarding
+Tilt can automatically create a port forward for the application using `k8s_resource`. 
+```python
+k8s_resource(
+    'carvel-tilt-demo', 
+     port_forwards=port
+ )
+```
+### Running Tilt
+
+Make sure the `REGISTRY` environment variable is set and simply run `tilt up`, this will read the `Tiltfile` and start the build/deploy workflow
+Once `tilt` is running, you can press the `s` key to stream the logs to the console or open [http://localhost:10350](http://localhost:10350) in your browser
+
+```shell
+# Set the variable to a registry you have push access to
+export REGISTRY=docker.io/myregistry
+
+# Build and deploy the project
+tilt up
+
+# Tear everything down
+tilt down
+```
+You can see the complete Tiltfile here [https://github.com/ojhughes/carvel-tilt-example-go/blob/main/Tiltfile](https://github.com/ojhughes/carvel-tilt-example-go/blob/main/Tiltfile)
+
+### See it in action!
+![](../../static/images/blog/tilt-animated-demo.gif)
```

Comment:
I think the gif does not show up because the path is malformed.
The path would be simply,
```suggestion
![](/images/blog/tilt-animated-demo.gif)
```

Reply:
yes that fixed it, thanks!